src/Documenter.jl

```diff
@@ -462,6 +465,11 @@ function deploydocs(;
                     # Remove the unencrypted private key.
                     isfile(keyfile) && rm(keyfile)
                 end
+
+                # Undo ssh config changes.
```

Comment:
Couldn't seem to get `GIT_SSH` working correctly with `withenv`, but did manage to remove the whole external shell script hack so I think this is as simple as it's going to get until the older authentication methods are finally removed.